rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,190 @@
+            )
+
+        if owner != o:
+            self.output.add_info(
+                'E',
+                pkg,
+                'permissions-incorrect-owner',
+                '%(file)s belongs to %(owner)s but should be %(o)s'
+                % {'file': path, 'owner': owner, 'o': o},
+            )
+
+    def _check_post_scriptlets(self, pkg, path, need_verifyscript):
+        script = pkg[rpm.RPMTAG_POSTIN] or pkg.scriptprog(rpm.RPMTAG_POSTINPROG)
+        found = False
+        need_set_permissions = False
+
+        if script:
+            for line in script.split('\n'):
+                if 'chkstat -n' in line and path in line:
+                    found = True
+                    break
+
+        if need_verifyscript and (
+            path not in self.perms or not self._is_static_entry(self.perms[path])
+        ):
+
+            if not script or not found:
+                self.output.add_info(
+                    'E',
+                    pkg,
+                    'permissions-missing-postin',
+                    'missing %%set_permissions %s in %%post' % path,
+                )
+
+            need_set_permissions = True
+            script = (
+                pkg[rpm.RPMTAG_VERIFYSCRIPT] or pkg[rpm.RPMTAG_VERIFYSCRIPTPROG]
+            )
+
+            found = False
+            if script:
+                for line in script.split('\n'):
+                    if '/chkstat' in line and path in line:
+                        found = True
+                        break
+
+            if not script or not found:
+                self.output.add_info(
+                    'W',
+                    pkg,
+                    'permissions-missing-verifyscript',
+                    'missing %%verify_permissions -e %s' % path,
+                )
+
+        return need_set_permissions
+
+    def _exists_permissions_entry(self, path, mode):
+        return path in self.perms or (stat.S_ISDIR(mode) and path + '/' in self.perms)
+
+    def _is_static_entry(self, entry):
+        # entries coming from the fixed permissions profile are considered
+        # static
+        return entry.profile.endswith('/permissions')
+
+    @staticmethod
+    def _paths_to(*file_names):
+        # we used to store the permissions data in /etc even though they aren't configuration files
+        # the whitelisting should check both paths (old /etc and new /usr/share) until all
+        # distributions using the old one (SLE15) are retired
+        for name in file_names:
+            # return the new path first.
+            # chkstat prefers the new paths over the old ones, so callers that only care about the
+            # first matching file must mimic that.
+            yield '/usr/share/permissions/' + name
+            yield '/etc/' + name
+
+    def check(self, pkg):
+        if pkg.is_source:
+            return
+
+        need_set_permissions = False
+
+        for f, pkgfile in pkg.files.items():
+
```

Comment:
Please remove empty lines at the beginning of a class definition, `for` and `if` block ..